src/python/pants/backend/javascript/subsystems/nodejs.py

```diff
@@ -390,7 +391,11 @@ class NodeJSBootstrap:
 async def _get_nvm_root() -> str | None:
     """See https://github.com/nvm-sh/nvm#installing-and-updating."""
 
-    env = await Get(EnvironmentVars, EnvironmentVarsRequest(("NVM_DIR", "XDG_CONFIG_HOME", "HOME")))
+    env = await environment_vars_subset(
```

Comment:
I used this call-site as proof, this routine is explicitly tested in `nodejs_test.py`

Reply:
This PR sounds good (make the migrations as incremental as possible)...

However, given `nodejs_test.py` has no changes, that'd imply to me `MockGet` is having to do all the solving that we're trying to move away from. Do you know if there's a "proper" call-by-name way to do mocking, for new code?

Reply:
MockGet is a much dumber resolution algorithm. It essentially does return type and arg type pattern matching to select a mock, there's no graph solving.

Regular module patching should work since its a function call now? E.g pytest-mock or other monkeypatching.